controllers/virtualmachine/volume/volume_controller.go

```diff
@@ -690,23 +693,14 @@ func (r *Reconciler) processAttachments(
 			}
 		}
 
-		// If we're allowing only one pending attachment, we cannot create the next CnsNodeVmAttachment
-		// until the previous ones are attached. This is really only effective when the VM is first being
-		// created, since the volumes could be added anywhere or the same ones reordered in the Spec.Volumes.
-		if onlyAllowOnePendingAttachment && hasPendingAttachment {
-			// Do not create another CnsNodeVmAttachment while one is already pending, but continue
-			// so we build up the Volume Status for any existing volumes.
-			continue
-		}
-
 		// If VM hardware version doesn't meet minimal requirement, don't create CNS attachment.
 		// We only fetch the hardware version when this is the first time to create the CNS attachment.
 		//
 		// This HW version check block is along with removing VMI hardware version check from VM validation webhook.
 		// We used to deny requests if a PVC is specified in the VM spec while VMI hardware version is not supported.
 		// In this case, no attachments can be created if VM hardware version doesn't meet the requirement.
 		// So if a VM has volume attached, we can safely assume that it has passed the hardware version check.
-		if len(volumeStatuses) == 0 && len(attachments) == 0 {
+		if checkHWVersion {
```

Comment:
JW, are you initializing the `checkHwVersion` variable for readability so that it is clear why we are doing this?  Cause I would imagine the comment block explaining this with the length check.

Reply:
This only needs to be done once. With things being shuffled around, the `len(volumeStatuses) == 0` may no longer be true despite already doing this check